src/hooks/useTransactionType.ts

```diff
@@ -84,20 +85,14 @@ const getTxType = (tx: TransactionSummary): TxType => {
 
       return {
         icon: toAddress?.logoUri || '/images/transactions/custom.svg',
-        text: toAddress?.name || 'Contract interaction',
+        text: addressBookName || toAddress?.name || 'Contract interaction',
       }
     }
     default: {
       return {
         icon: '/images/transactions/custom.svg',
-        text: 'Contract interaction',
+        text: addressBookName || 'Contract interaction',
       }
     }
   }
 }
-
-export const useTransactionType = (tx: TransactionSummary): TxType => {
-  return useMemo(() => {
-    return getTxType(tx)
-  }, [tx])
-}
```

Comment:
won't we lose performance for dismissing this memoization?

Reply:
The performance increased seemed neglibile when looking at the dependency array but I'll put it back.